cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -31,7 +31,7 @@ pub enum PublishError {
     UnsupportedEnvironmentState(String),
 
     #[error("There was an error communicating with the catalog")]
-    CatalogError(#[source] Box<dyn error::Error>),
+    CatalogError(#[source] Box<dyn error::Error + Send>),
```

Comment:
suggestion: also require `Sync` bound

`Send` implies that you can safely _move_ objects of this type to other threads (or out of other threads). In this case that means, you can use the error produced in the closure of the Spinner (which runs on a separate thread) on the main thread after the spinning operation completed.

`Sync` implies that you can share _references_ (or borrows) of the object as well.

Usually you want both if possible, and especially for errors this is usually true as errors should be immutable. `Send + Sync` is true for most types in rust anyway unless the type intentionally evades the ownership model (reference counted values, cell types that allow mutating their value despite not being mutable themselve, etc.)


```suggestion
    CatalogError(#[source] Box<dyn error::Error + Send + Sync>),
```

Reply:
Thanks, that's _really_ helpful context!